tests/realtikvtest/importintotest4/cloud_sdk_test.go

```diff
@@ -164,3 +170,102 @@ func (s *mockGCSSuite) TestDumplingSource() {
 
 	s.Require().NoError(mock.ExpectationsWereMet())
 }
+
+func (s *mockGCSSuite) TestAutoDetectFileType() {
+	s.server.CreateBucketWithOpts(fakestorage.CreateBucketOpts{Name: "auto_detect"})
+
+	items := []struct {
+		name       string
+		buf        func() []byte
+		expectRows []string
+	}{
+		{name: "noext", buf: func() []byte { return []byte("1,foo\n2,bar\n") }, expectRows: []string{"1 foo", "2 bar"}},
+		// SQL data but no suffix -> used for negative case
+		{name: "sql_noext", buf: func() []byte { return []byte("INSERT INTO auto_detect.t VALUES (13,'m'),(14,'n');\n") }, expectRows: nil},
+		// CSV data but with .sql suffix -> should be detected as SQL and trigger CSV-option errors
+		{name: "csv_as_sql.sql", buf: func() []byte { return []byte("15,p\n16,q\n") }, expectRows: nil},
+		{name: "f1.CSV", buf: func() []byte { return []byte("3,baz\n4,qux\n") }, expectRows: []string{"3 baz", "4 qux"}},
+		{name: "data.sql", buf: func() []byte { return []byte("INSERT INTO auto_detect.t VALUES (5,'e'),(6,'f');\n") }, expectRows: []string{"5 e", "6 f"}},
+		{name: "p.parquet", buf: func() []byte { return []byte{} }, expectRows: nil},
```

Comment:
we can generate a small parquet file and store it under this folder and load it for testing